resources/migrations/001_update_migrations.yaml

```diff
@@ -8253,6 +8253,152 @@ databaseChangeLog:
             tableName: login_history
             indexName: idx_user_id_device_id
 
+  - changeSet:
+      id: v51.2024-06-13T11:22:39
+      author: johnswanson
+      comment: Add `last_viewed_at` to dashboards
+      changes:
+        - addColumn:
+            tableName: report_dashboard
+            columns:
+              - column:
+                  name: last_viewed_at
+                  type: ${timestamp_type}
+                  remarks: Timestamp of when this dashboard was last viewed
+                  constraints:
+                    nullable: true
+
+  - changeSet:
+      id: v51.2024-06-14T06:45:34
+      author: johnswanson
+      comment: Create index on view_log (model, model_id, timestamp)
```

Comment:
Is this index just to make the subsequent migration more efficient or is there another reason for it? Just for my own understanding.

Reply:
yea I'd like to understand more about the motivation here. 

IIRC view_log is unbounded, so this migration will take a lot of time, plus it'll create overhead for inserting, which we do quite frequently in a synchronous fashion.

Reply:
So yeah, it is to make the migration more efficient. Testing on data from stats (2.8M records in the view_log, 1650 dashboards), I found that without the index, the subsequent migration took ~30s. Creating the index took 10s and sped up the subsequent migration to ~200ms. I informally tested a few inserts into the view log before/after and the additional overhead wasn't detectable beneath the random variation in timing. Of course, if we do want to avoid that overhead I'm happy to delete the index in a subsequent migration.

Reply:
👍 to skip the index, I don't think it's worth the trouble here given how little difference it makes